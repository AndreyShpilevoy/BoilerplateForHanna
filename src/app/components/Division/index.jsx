import React from 'react';
import {string, shape, number} from 'prop-types';

const Division = ({className, division}) => {
    const {id, title, HID} = division;
    const userInfoString = `Division id: ${id}, title: ${title}, HID: ${HID}`;
    return (
        <div className={className}>
            {userInfoString}
        </div>
    );
};

Division.propTypes = {
    className: string,
    division: shape({
        id: number.isRequired,
        title: string.isRequired,
        HID: number.isRequired
    }).isRequired
};

Division.defaultProps = {
    className: ''
};

export default Division;
