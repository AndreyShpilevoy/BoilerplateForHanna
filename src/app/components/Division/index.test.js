/* eslint-disable no-undef, max-statements, import/first, react/jsx-filename-extension */

import React from 'react';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Division from './index';

configure({adapter: new Adapter()});

describe('Division', () => {
    it('Division match expected snapshot', () => {
        const division = {id: 1, title: 'company', HID: 12};
        expect(shallow(<Division className='division-class-name' division={division} />)).toMatchSnapshot();
    });
});
