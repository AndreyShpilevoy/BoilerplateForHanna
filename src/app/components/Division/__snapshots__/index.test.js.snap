// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Division Division match expected snapshot 1`] = `
<div
  className="division-class-name"
>
  Division id: 1, title: company, HID: 12
</div>
`;
